packages/itwin/grouping-mapping-widget/src/widget/components/hooks/useMappingsOperations.ts

```diff
@@ -30,22 +31,82 @@ const fetchMappings = async (
   }
 };
 
+const fetchExtractionStatus = async (
+  iModelId: string,
+  getAccessToken: GetAccessTokenFn,
+  extractionClient: IExtractionClient,
+  setIconStatus: React.Dispatch<React.SetStateAction<"negative" | "positive" | "warning">>,
+  setIconMessage: React.Dispatch<React.SetStateAction<string>>,
+  setExtractionMessageData: React.Dispatch<React.SetStateAction<ExtractionMessageData[]>>
+) => {
+  setIconStatus("warning");
+  setIconMessage("Extraction status pending.");
+  try {
+    const accessToken = await getAccessToken();
+    const extractions = await extractionClient.getExtractionHistory(accessToken, iModelId);
+    const jobId = extractions[0].jobId;
+    const status = await extractionClient.getExtractionStatus(accessToken, jobId);
+    if (status.containsIssues) {
+      setIconStatus("negative");
+      setIconMessage("Extraction contains issues. Click to view extraction logs.");
+      let logs = await extractionClient.getExtractionLogs(accessToken, jobId);
+      logs = logs.filter((log) => log.message != null);
+      const extractionMessageData = logs.map((log) =>
+        (
+          {
+            date: log.dateTime,
+            category: log.category,
+            level: log.level,
+            message: String(log.message),
+          }
+        ));
+      setExtractionMessageData(extractionMessageData);
+    } else {
+      setIconStatus("positive");
+      setIconMessage("Extraction Successful.");
+    }
+  } catch (error: any) { }
```

Comment:
What is the IconStatus or IconMessage when error occurs? Let's say the `getExtractionLogs` fails for some reason. The icon status will stay negative but you will have no data to display.

Reply:
I have updated this part.